ams/server/src/main/java/com/netease/arctic/server/dashboard/controller/TableController.java

```diff
@@ -149,19 +152,25 @@ public void getHiveTableDetail(Context ctx) {
             && StringUtils.isNotBlank(db)
             && StringUtils.isNotBlank(table),
         "catalog.database.tableName can not be empty in any element");
+    ServerCatalog serverCatalog = tableService.getServerCatalog(catalog);
+    HMSClientPool hmsClientPool = null;
+    if (serverCatalog instanceof MixedHiveCatalogImpl) {
+      MixedHiveCatalogImpl mixedHiveCatalog = (MixedHiveCatalogImpl) serverCatalog;
+      hmsClientPool = mixedHiveCatalog.getHiveClient();
+    } else if (serverCatalog instanceof ExternalCatalog) {
+      ExternalCatalog externalCatalog = (ExternalCatalog) serverCatalog;
+      hmsClientPool = externalCatalog.getHMSClientPool();
+    }
```

Comment:
Maybe we can expose `public getTableMetaStore` in `ExternalCatalog` and build HMSClient in this Controller. This will make the code more versatile

Reply:
> Maybe we can expose `public getTableMetaStore` in `ExternalCatalog` and build HMSClient in this Controller. This will make the code more versatile

Good suggestion! i will modify the implementation.